### PR TITLE
Add JFR InitialEnvironmentVariable support

### PR DESCRIPTION
**Pull request by thallium**

### CODE REVIEW

**Comment by thallium** [comment]
@tajila FYI

File: runtime/vm/JFRChunkWriter.cpp [hunk 1]
```diff
@@ -189,6 +189,16 @@ VM_JFRChunkWriter::writeStringLiteral(const char *string)
 	}
 }
 
+void
+VM_JFRChunkWriter::writeStringLiteral(const char *string, UDATA len)
```
Comment (tajila) [inline runtime/vm/JFRChunkWriter.cpp:193]: instead you can just use `writeUTF8String` with the approriate cast

File: runtime/vm/JFRChunkWriter.cpp [hunk 2]
```diff
@@ -815,4 +825,48 @@ VM_JFRChunkWriter::writeInitialSystemPropertyEvents(J9JavaVM *vm)
...
+{
+	J9SysinfoEnvIteratorState envState;
+	memset(&envState, 0, sizeof(envState));
+
+	/* Call init with zero length buffer to get the required buffer size */
+	int32_t result = j9sysinfo_env_iterator_init(&envState, NULL, 0);
+
+	if (result >= 0) {
+		int32_t bufferSize = result;
+		void *buffer = j9mem_allocate_memory(bufferSize, OMRMEM_CATEGORY_VM);
```
Comment (tajila) [inline runtime/vm/JFRChunkWriter.cpp:839]: need to free this memory

File: runtime/vm/JFRChunkWriter.cpp [hunk 3]
```diff
@@ -815,4 +825,48 @@ VM_JFRChunkWriter::writeInitialSystemPropertyEvents(J9JavaVM *vm)
...
+
+		if (NULL != buffer) {
+			J9SysinfoEnvElement envElement = {NULL};
+
+			result = j9sysinfo_env_iterator_init(&envState, buffer, bufferSize);
```
Comment (tajila) [inline runtime/vm/JFRChunkWriter.cpp:844]: need to check if `if (result < 0) {` if so there was a problem getting the env vars

Comment (tajila) [reply runtime/vm/JFRChunkWriter.cpp:844]: Oh okay, miseed that. Leave it as is

Comment (tajila) [reply runtime/vm/JFRChunkWriter.cpp:844]: Actaully, sorry, an error can still occur even if the buffer is valid, so please add the check

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
jenkins test sanity.functional xlinux jdk17

**Pull request merged**